compiled_starters/redis-starter-java/spawn_redis_server.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+set -e
+mkdir -p out
+javac -sourcepath src src/Main.java -d out
+jar cf java-redis.jar -C out/ .
+rm -rf out
+java -cp java-redis.jar Main "$@"
```

Comment:
To ensure that signals are passed properly to the child process, we'll need to prepend `exec` to the last command here.

Example: https://github.com/codecrafters-io/languages/blob/5746db439eae64c99189ce464509f0d4a747f2b1/compiled_starters/git-starter-ruby/your_git.sh#L2

(Just realizing that this is something we could add to the contributing guide 🙂) 

Also, can we change `out` to a temporary directory created with `mktemp -d`? This way, even if the command fails halfway, a user won't end up with the `out` directory polluting their git status. This is assuming that the directory doesn't contain any human-readable stuff. If it does, then maybe sticking with `out` is fine 👍